models/error.go

```diff
@@ -674,6 +674,23 @@ func (err ErrRepoAlreadyExist) Error() string {
 	return fmt.Sprintf("repository already exists [uname: %s, name: %s]", err.Uname, err.Name)
 }
 
+// ErrForkAlreadyExist represents a "ForkAlreadyExist" kind of error.
+type ErrForkAlreadyExist struct {
+	Uname    string
+	RepoName string
+	ForkName string
+}
+
+// IsErrForkAlreadyExist checks if an error is an ErrForkAlreadyExist.
+func IsErrForkAlreadyExist(err error) bool {
+	_, ok := err.(ErrForkAlreadyExist)
+	return ok
+}
+
+func (err ErrForkAlreadyExist) Error() string {
+	return fmt.Sprintf("repository is already forked [uname: %s, repo path: %s, fork path: %s]", err.Uname, err.RepoName, err.ForkName)
```

Comment:
```suggestion
	return fmt.Sprintf("Fork of repository already exists, only one fork per user allowed [uname: %s, repo path: %s, fork path: %s]", err.Uname, err.RepoName, err.ForkName)
```
Think this makes it a bit more clear, since part of what made the original issue unclear was the assumption that you can fork multiple times

Reply:
Small nit, can you change `Fork` to lowercase? 
I think it's a bit verbose for an error message, but it does clarify it so I'm not too hung up on it.

Reply:
Sure, I think that suggestion is fine too! Makes is more clear that the error is because a user already has a fork, and not that the repo has been forked in general